### PR TITLE
OCPBUGS-22348: address CVE-2023-44487

### PR DESCRIPTION
**Pull request by alebedev87**
Backport of https://github.com/openshift/external-dns-operator/pull/204:
- bump controller-runtime v0.13.2
- bump the operand image to [the latest 1.1 version](https://quay.io/repository/external-dns-operator/external-dns/manifest/sha256:fb8ea3784012a21a1c679c23cd1d4a2dbf499d00517bcfe0c70cb85cd52dd290) (`fb8ea37`)
- disable HTTP/2 for kube-rbac-proxy container
- disable HTTP/2 for webhook server

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -38,14 +38,15 @@ spec:
...
+        - --webhook-disable-http2
         env:
         - name: OPERATOR_NAMESPACE
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
         - name: RELATED_IMAGE_EXTERNAL_DNS
-          # openshift/external-dns commit: fe00b4b83c2263282a9068655e8e3fbbc167b653
-          value: quay.io/external-dns-operator/external-dns@sha256:a742a727d3a18ffb78c7d0a24fd2c5eebed01479163959169203427514d7aa89
+          # openshift/external-dns commit: 0e095a3db5ae9518a1744dd85d3f7ed2e359768d
```
Comment (gcs278) [inline config/manager/manager.yaml:48]: Double check my work, but I think this is wrong:
```suggestion
          # openshift/external-dns commit: 98c4b73979743c7a43a29e518232eada37ea64f5
```
Source `vcs-ref`: https://quay.io/repository/external-dns-operator/external-dns/manifest/sha256:fb8ea3784012a21a1c679c23cd1d4a2dbf499d00517bcfe0c70cb85cd52dd290

**Review by gcs278 — COMMENTED** [review]
One nit pick; otherwise looks good.

File: config/manager/manager.yaml (hunk 1 above)
Comment (alebedev87) [reply config/manager/manager.yaml:48]: It is, a wrong copy paste.

**Comment by gcs278** [comment]
/lgtm
@miciah want to approve?

**Comment by Miciah** [comment]
Thanks!
/approve

**Pull request merged**